build/RestoreDependency.proj

```diff
@@ -8,13 +8,22 @@
 
   <UsingTask TaskName="DotNetRestore" AssemblyFile="$(CLIBuildDllPath)" />
 
+<!-- 
+
+  Note that the CopySdkToOutput target explicitly does NOT have Inputs and Outputs declared.
+  This is because it is supposed to overlay updated SDK items on a folder that may already
+  have different versions from a different package.  If Inputs and Outputs are declared,
+  MSBuild will skip files where a newer version already exists in the target folder, which
+  is not what we want.
+  
+ -->
+
   <Target Name="CopySdkToOutput"
           DependsOnTargets="PrepareBundledDependencyProps;
                             EnsureDependencyRestored;
                             GetSdkItemsToCopy"
           Condition="'$(SdkLayoutDirectory)' != ''"
-          Inputs="@(SdkContent)"
-          Outputs="@(SdkContent->'$(SdkLayoutDirectory)/%(RecursiveDir)%(FileName)%(Extension)')">
+>
```

Comment:
Should we be using the arg to preserve newest here given that we can't use inputs/outputs.

Reply:
The issue is we don't actually want to preserve newest here.  There may be newer versions of the files from a different source that we want to overlay with the right ones.

There's probably a better way to do this (perhaps exclude the SDK files from the initial files that are copied to the layout), but I didn't get into that in this PR.